From 94c34010c2063eafdeac2032ff7dc9a70e65f130 Mon Sep 17 00:00:00 2001
From: Jiang Lu <lu.jiang@windriver.com>
Date: Tue, 16 Aug 2011 15:50:24 +0800
Subject: [PATCH 08/10] P1021MDS:ATM:ATM configuration for P1021MDS

This patch adds initial configuration for ATM feature on P1021MDS board,
including bcsr and PMUXCR settings.

Signed-off-by: Liu Yu <yu.liu@freescale.com>

[Extracted from vendor drop QorIQ-NONDPAA-SDK-V1-20110429_ltib.iso
linux-2.6.35-qoriq-P1021MDS-ATM-ATM-configuration-for-P102.patch
Fixed context to apply to Wind River kernel tree.]

Integrated-by: Jiang Lu <lu.jiang@windriver.com>
---
 arch/powerpc/platforms/85xx/mpc85xx_mds.c |   17 +++++++++++++++++
 1 files changed, 17 insertions(+), 0 deletions(-)

diff --git a/arch/powerpc/platforms/85xx/mpc85xx_mds.c b/arch/powerpc/platforms/85xx/mpc85xx_mds.c
index f111013..3f1ab73 100644
--- a/arch/powerpc/platforms/85xx/mpc85xx_mds.c
+++ b/arch/powerpc/platforms/85xx/mpc85xx_mds.c
@@ -305,6 +305,13 @@ static void __init mpc85xx_mds_setup_arch(void)
 			clrbits8(&bcsr_regs[11], BCSR11_ENET_MICRST);
 			setbits8(&bcsr_regs[11], BCSR11_ENET_MICRST);
 
+#if defined CONFIG_ATM_UCC
+#define BCSR6_LBC_UPC          (0x1 << 5)
+#define BCSR6_UPC_EN           (0x1 << 4)
+			clrbits8(&bcsr_regs[6], BCSR6_LBC_UPC);
+			clrbits8(&bcsr_regs[6], BCSR6_UPC_EN);
+#endif
+
 #ifdef CONFIG_TDM
 			clrbits8(&bcsr_regs[6], BCSR6_1588_TDMA);
 			setbits8(&bcsr_regs[6], BCSR6_TDMB_ENET1 |
@@ -337,6 +344,16 @@ static void __init mpc85xx_mds_setup_arch(void)
 					MPC85xx_PMUXCR_QE3);
 #endif /* CONFIG_UCC_GETH */
 
+#if defined CONFIG_ATM_UCC
+			clrbits32(pmuxcr, MPC85xx_PMUXCR_QE12);
+			clrbits32(pmuxcr, MPC85xx_PMUXCR_QE9);
+			setbits32(pmuxcr, MPC85xx_PMUXCR_QE0 |
+					  MPC85xx_PMUXCR_QE2 |
+					  MPC85xx_PMUXCR_QE4 |
+					  MPC85xx_PMUXCR_QE5 |
+					  MPC85xx_PMUXCR_QE11);
+#endif
+
 #ifdef CONFIG_TDM
 
 			/* Clear QE12 for releasing the LBCTL */
-- 
1.7.0.2

